test/run.py

```diff
@@ -0,0 +1,248 @@
+import argparse
+import multiprocessing
+import os
+import re
+import signal
+import sys
+from glob import glob
+from typing import List, Optional, Set
+
+import test_utils
+from run_end_to_end_test import run_end_to_end_test
+from test_utils import CLI_BUILDER_IMAGE, CLI_IMAGE, \
+    CONTROLLER_CONTAINER, CONTROLLER_HOSTNAME, CONTROLLER_IMAGE, \
+    CONTROLLER_PORT, CONTROLLER_TESTS_CONTAINER, CONTROLLER_TESTS_IMAGE, \
+    DATA_DIRECTORY, PLZ_ROOT_DIRECTORY, PLZ_USER, TEST_DIRECTORY, \
+    docker_compose, get_network
+
+
+def start_controller():
+    test_utils.print_info('Building the controller...')
+    test_utils.execute_command([
+        'docker',
+        'image',
+        'build',
+        '--quiet',
+        f'--tag={CONTROLLER_IMAGE}',
+        os.path.join(PLZ_ROOT_DIRECTORY, 'services/controller')])
```

Comment:
Why not use [docker-py](https://docker-py.readthedocs.io/) instead of shelling out to Docker everywhere?